.github/workflows/code-quality.yml

```
@@ -0,0 +1,345 @@
+          cache-dependency-path: services/shared-types/package-lock.json
+
+      - name: 📥 Install shared-types dependencies
+        working-directory: services/shared-types
+        run: npm ci
+
+      - name: 🏗️ Build shared-types
+        working-directory: services/shared-types
+        run: npm run build
+
+      - name: 📦 Upload shared-types build artifacts
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.7)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)